packages/core/src/hooks/resource/useResource/index.ts

```diff
@@ -1,20 +1,47 @@
 import { useContext } from "react";
-import { ResourceContext } from "@contexts/resource";
-import { IResourceContext } from "../../../contexts/resource/IResourceContext";
 
-export type UseResourceType = {
-    (): {
-        resources: IResourceContext["resources"];
-    };
+import { ResourceContext, IResourceContext } from "@contexts/resource";
+import { BaseKey, IResourceItem, ResourceRouterParams } from "src/interfaces";
+import { useRouterContext, useResourceWithRoute } from "@hooks";
+
+export type UseResourcePropsType = {
+    resourceName?: string;
```

Comment:
```suggestion
    resourceName?: string;
```
resourceName must be deprecated